### PR TITLE
Issue #1299 public api fixes from imod5

### LINKED ISSUE #1299: Final public API changes to ``from_imod5_data`` before tutorial
There are the following issues with the public API for ``from_imod5_data`` in the 0.18.0 release:

- The ``SimulationAllocationOptions`` and ``SimulationDistributingOptions`` are stowed away very far, and not part of the public API
- Yet, these objects used to store default settings are required to call the public api method ``Modflow6Simulation.from_imod5_data``...
- A description of "period_times" is missing in the docstring
- Example code snippets are missing from the docstring

### PR DESCRIPTION
**Pull request by JoerivanEngelen**
Fixes #1299

* The SimulationAllocationOptions and SimulationDistributingOptions are stowed away very far, and not part of the public API
* Yet, these objects used to store default settings are required to call the public api method Modflow6Simulation.from_imod5_data...
* A description of "period_times" is missing in the docstring
* Example code snippets are missing from the docstring

# Description
This PR fixes the following things:
* Makes ``SimulationAllocationOptions`` and ``SimulationDistributingOptions`` part of the public API
* Makes ``allocation_options`` and ``distributing_options`` optional arguments for  ``Modflow6Simulation.from_imod5_data`` and ``GroundwaterFlowModel.from_imod5_data``
* To do this, I had to change the order of arguments for Modflow6Simulation.from_imod5_data, as Python doesn't allow non-optional args after an optional arg. This can break existing scripts, however this has just been released, so not many scripts use it. As long as the tutorial material is updated within time.
* Extends docstrings of ``GroundwaterFlowModel.from_imod5_data``, ``Modflow6Simulation.from_imod5_data``, ``SimulationAllocationOptions``, and ``SimulationDistributingOptions``

# Checklist

- [x] Links to correct issue
- [x] Update changelog, if changes affect users
- [x] PR title starts with ``Issue #nr``, e.g. ``Issue #737``
- [ ] Unit tests were added
- [ ] **If feature added**: Added/extended example

### CODE REVIEW

File: imod/mf6/simulation.py [hunk 1]
```diff
@@ -1331,51 +1331,88 @@ def mask_all_models(
...
+        regridder_types: Optional[dict[str, RegridMethodType]] = None,
     ) -> "Modflow6Simulation":
         """
-        Imports a GroundwaterFlowModel (GWF) from the data in an IMOD5 project file.
-        It adds the packages for which import from imod5 is supported.
-        Some packages (like OC) must be added manually later.
-
+        Imports a GroundwaterFlowModel (GWF) from the data in an IMOD5 project
+        file and puts it in a simulation. It adds the packages for which import
+        from imod5 is supported. Some packages (like OC) must be added manually
```
Comment (Manangka) [inline imod/mf6/simulation.py:1343]: `It adds the packages for which import  from imod5 is supported.`
This sentence is a bit unclear/ could some more explaination.

Which packages are supported? Do some packages have a `from_imod5_data` method? How do i now which packages are imported?

File: imod/mf6/simulation.py [hunk 2]
```diff
@@ -1331,51 +1331,88 @@ def mask_all_models(
...
+        later.
```
Comment (Manangka) [inline imod/mf6/simulation.py:1344]: `Some packages (like OC) must be added manually later`
what about:
`Some packages (like OC) won't be imported automatically and have to be imported manually .`

File: imod/mf6/simulation.py [hunk 3]
```diff
@@ -1331,51 +1331,88 @@ def mask_all_models(
...
-            then it should be imported separately
+            under the key of the package type to which it belongs.
+        period_data: dict[str, list[datetime]]
+            dictionary containing the package names mapped to a list of repeated
+            stress periods. These are set as ``repeat_stress``.
         times:  list[datetime]
-            time discretization of the model to be imported.
+            time discretization of the model to be imported. This is used for two things:
+                * Times of wells with associated timeseries are resampled to these times
+                * Start- and end times in the list are used to repeat the stresses
```
Comment (Manangka) [inline imod/mf6/simulation.py:1357]: `Start- and end times` -> `Start and end times`

**Review by Manangka — APPROVED** [review]

File: imod/mf6/simulation.py (hunk 1 above)
Comment (JoerivanEngelen) [reply imod/mf6/simulation.py]: Thanks, I think we need some kind of list somewhere to add what iMOD5 functionality is supported and what is not yet.

**Pull request merged**